CONTRIBUTING.md

```diff
@@ -1,28 +1,48 @@
 # Contributing Guide
 
-First off, thanks for taking the time to contribute!
+Thank you for taking the time to contribute to Iroha 2!
 
-The following is a short set of guidelines for contributing to Iroha.
+Please read this guide to learn how you can contribute and which guidelines we expect you to follow. This includes the guidelines about code and documentation as well as our conventions regarding git workflow.
+
+Reading these guidelines will save you time later.
 
 ## How Can I Contribute?
 
+There are a lot ways you could contribute to our project:
+
+- Report [bugs](#reporting-bugs) and [vulnerabilities](#reporting-vulnerabilities)
+- [Suggest improvements](#suggesting-improvements) and implement them
+- [Ask questions](#asking-questions) and engage with the community
+
+New to our project? [Make your first contribution](#your-first-code-contribution)!
+
 ### TL;DR
 
 * Find [ZenHub](https://app.zenhub.com/workspaces/iroha-v2-60ddb820813b9100181fc060/board?repos=181739240).
 * Fork [Iroha](https://github.com/hyperledger/iroha/tree/iroha2-dev).
 * Fix your issue of choice.
+* Ensure you follow our [style guides](#style-guides) for code and documentation.
 * Write [tests](https://doc.rust-lang.org/cargo/commands/cargo-test.html). Ensure they all pass (`cargo test`).
-* Fix [`clippy`](https://lib.rs/crates/cargo-lints) warnings: `cargo lints clippy --workspace --benches --tests --examples --all-features`
-* Format code `cargo +nightly fmt --all` and generate docs `cargo run --bin kagami -- docs >"docs/source/references/config.md" && git add "docs/source/references/config.md"`
-* `git pull -r hyperledger iroha2-dev`, `git commit -s`, `git push <your-fork>`, and [create a pull request](https://github.com/hyperledger/iroha/compare) to the `iroha2-dev` branch on GitHub.
+* Fix [`clippy`](https://lib.rs/crates/cargo-lints) warnings: `cargo lints clippy --workspace --benches --tests --examples --all-features`.
+* Format code `cargo +nightly fmt --all` and generate docs `cargo run --bin kagami -- docs >"docs/source/references/config.md" && git add "docs/source/references/config.md"`.
+* `git pull -r hyperledger iroha2-dev`, `git commit -s`, `git push <your-fork>`, and [create a pull request](https://github.com/hyperledger/iroha/compare) to the `iroha2-dev` branch on GitHub. Ensure the PR has the `[type] #<issue number>: Description` [title](#pull-request-titles).
```

Comment:
Should probably be `git pull -r upstream iroha2-dev` here. Could be beneficial to explain how to set upstream to track tho original repo for your fork as well, but maybe not in a tl;dr.

Reply:
Depends. I call it `upstream` for uniformity. `hyperledger` is less ambiguous.

Reply:
> Could be beneficial to explain how to set upstream to track the original repo for your fork as well, but maybe not in a tl;dr section.

I can add links to github docs on this: https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/configuring-a-remote-for-a-fork
I don't think we need to add these instructions here

Reply:
> Depends. I call it `upstream` for uniformity. `hyperledger` is less ambiguous.

<img width="641" alt="image" src="https://user-images.githubusercontent.com/17865025/169018571-75399c8c-f623-4be7-96ed-773801be5c88.png">

what about this?